expression/builtin_compare_vec.go

```diff
@@ -590,11 +592,78 @@ func (b *builtinNullEQStringSig) vecEvalInt(input *chunk.Chunk, result *chunk.Co
 }
 
 func (b *builtinLTIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinLTIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	var err error
+	if err = b.args[0].VecEvalInt(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	return VecCompareInt(b.args[0], b.args[1], result, buf, resOfLT, result)
+}
+
+// VecCompareInt compares two integers.
+func VecCompareInt(lhsArg, rhsArg Expression, lColumn, rColumn *chunk.Column, f func(val int64, isNull bool, err error) (int64, bool, error), result *chunk.Column) error {
+	arg0 := lColumn.Int64s()
+	arg1 := rColumn.Int64s()
+	result.MergeNulls(lColumn, rColumn)
+	i64s := result.Int64s()
+	n := len(arg0)
+	for i := 0; i < n; i++ {
+		var res int64
+		var isNull bool = false
+		var err error = nil
+		// compare null values.
+		if result.IsNull(i) {
+			result.SetNull(i, true)
+			res, isNull, err = f(compareNull(lColumn.IsNull(i), rColumn.IsNull(i)), false, nil)
+			if err != nil {
+				return err
+			}
+			i64s[i] = res
+			continue
+		}
+		isUnsigned0, isUnsigned1 := mysql.HasUnsignedFlag(lhsArg.GetType().Flag), mysql.HasUnsignedFlag(rhsArg.GetType().Flag)
+		switch {
```

Comment:
I think the performance would be better if we take the switch statement out from the fop loop.
You can take a look at #12559

Reply:
ok, i will try. Thank you